runtime/monovm-bridge.m

```diff
@@ -85,7 +82,36 @@
 	mono_install_unhandled_exception_hook (xamarin_unhandled_exception_handler, NULL);
 	mono_install_ftnptr_eh_callback (xamarin_ftnptr_exception_handler);
 
-	mono_jit_init_version ("MonoTouch", "mobile");
+#if DOTNET
+	bool use_mono_workaround = xamarin_init_mono_debug;
+#else
+	bool use_mono_workaround = false;
+#endif
+
+	const char *argc[] = {
+		"",
+		"--interp=-all"
+	};
+
+	if (use_mono_workaround) {
+		// This is a workaround for a runtime bug that prevents debugging from working properly.
+		// We call mono_main to disable interpreter optimizations when debugging, because
+		// mono's own interpreter initialization doesn't take into account that the debugger
+		// might be attached, and the subsequent optimizations can break the debugger.
+		// The stdout dance is to hide mono's "helpful" output telling us the command line
+		// arguments are incorrect (which they would be normally, but we're abusing mono_main
+		// to get the side effects).
+		fflush (stdout);
+		int originalStdout = dup (STDOUT_FILENO);
+		int devnull = open ("/dev/null", O_RDWR);
+		dup2 (devnull, STDOUT_FILENO);
+		mono_main (2, (char**)argc);
+		fflush (stdout);
+		dup2 (originalStdout, STDOUT_FILENO);
+	} else {
+		mono_jit_init_version ("MonoTouch", "mobile");
```

Comment:
I think we need this code here:
	if (xamarin_init_mono_debug)
		mono_debug_init (MONO_DEBUG_FORMAT_MONO);

Reply:
Ah yeah

Reply:
Fixed